greeting-service/src/main/java/dev/snowdrop/example/service/NameService.java

```diff
@@ -16,37 +16,47 @@
 
 package dev.snowdrop.example.service;
 
-import com.netflix.hystrix.HystrixCircuitBreaker;
-import com.netflix.hystrix.HystrixCommandKey;
-import com.netflix.hystrix.contrib.javanica.annotation.HystrixCommand;
-import com.netflix.hystrix.contrib.javanica.annotation.HystrixProperty;
+import io.github.resilience4j.circuitbreaker.CircuitBreaker;
+import io.github.resilience4j.circuitbreaker.CircuitBreakerRegistry;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpServerErrorException;
+import org.springframework.web.client.ResourceAccessException;
 import org.springframework.web.client.RestTemplate;
 
 /**
- * Service invoking name-service via REST and guarded by Hystrix.
+ * Service invoking name-service via REST and guarded by Resilience4j.
  */
 @Service
 public class NameService {
-
-    private static final HystrixCommandKey KEY = HystrixCommandKey.Factory.asKey("NameService");
-
     private final String nameHost = System.getProperty("name.host", "http://spring-boot-circuit-breaker-name:8080");
+    private static final String CIRCUIT_BREAKER_BACKEND = "nameService";
     private final RestTemplate restTemplate = new RestTemplate();
 
-    @HystrixCommand(commandKey = "NameService", fallbackMethod = "getFallbackName", commandProperties = {
-            @HystrixProperty(name = "execution.isolation.thread.timeoutInMilliseconds", value = "1000")
-    })
+    @Autowired
+    private CircuitBreakerRegistry circuitBreakerRegistry;
+
+    public NameService() {
+    }
+
+    @io.github.resilience4j.circuitbreaker.annotation.CircuitBreaker(name = CIRCUIT_BREAKER_BACKEND, fallbackMethod = "getFallbackName")
     public String getName() {
         return restTemplate.getForObject(nameHost + "/api/name", String.class);
     }
 
-    private String getFallbackName() {
+    public String getFallbackName(ResourceAccessException ex) {
         return "Fallback";
     }
 
+    public String getFallbackName() {
+        return "Fallback";
+    }
+
+    public String getFallbackName(RuntimeException ex) { return "Fallback"; }
+
+    public String getFallbackName(HttpServerErrorException ex) { return "Fallback"; }
```

Comment:
Why do we need four `getFallbackName` methods? I think the only needed is `public String getFallbackName(ResourceAccessException ex)`.

Reply:
I removed the one without parameters. The others are used on different tests.